base/intfuncs.jl

```diff
@@ -471,6 +469,27 @@ function ndigits0zpb(x::Base.BitUnsigned, b::Int)
     return d
 end
 
+function ndigits0zpb(x::Integer, b::Integer)
```

Comment:
What about, in order to avoid duplicating the algorithm, having the signature of the function be `function ndigits0zpb(x::Integer, b::Integer)`, condtionning the tests (`b==2`...) to `x isa `Base.Integer`, and add `x = unsigned(abs(x))` in this branch, and delete the following forwarding methods:
```
ndigits0zpb(x::Base.BitSigned, b::Integer) = ndigits0zpb(unsigned(abs(x)), Int(b))
ndigits0zpb(x::Base.BitUnsigned, b::Integer) = ndigits0zpb(x, Int(b))
```
(just an idea, if I'm motivated enough I could bother myself with it later)